### PR TITLE
Src: encodings_fragment -> colorspace_fragment

### PR DESCRIPTION
**Pull request by WestLangley**
... plus related changes.

### CODE REVIEW

**Comment by Mugen87** [comment]
I'm a bit worried about this change. The `encodings_fragment` shader chunk is frequently used in custom shader materials. With this change, all these material definitions will be broken.

So I wonder if the renaming is worth the breakage. 

I think it would be best to deprecate the shader chunk first and then log a warning for ten releases. Maybe we can try to implement a small deprecation check in `WebGLProgram` that checks for outdated chunks and performs a resolution to newer ones. Meaning logic that maps `encodings_fragment` to `color_space_fragment`.

**Comment by Mugen87** [comment]
This should do it:
```js
const shaderChunkMap = new Map( [
	[ 'encodings_fragment', 'color_space_fragment' ] // @deprecated, r154
] );

function includeReplacer( match, include ) {

	let string = ShaderChunk[ include ];

	if ( string === undefined ) {

		const newInclude = shaderChunkMap.get( include );

		if ( newInclude !== undefined ) {

			string = ShaderChunk[ newInclude ];
			console.warn( 'THREE.WebGLRenderer: Shader chunk "%s" has been deprecated. Use "%s" instead.', include, newInclude );

		} else {

			throw new Error( 'Can not resolve #include <' + include + '>' );

		}

	}

	return resolveIncludes( string );

}
```

**Comment by WestLangley** [comment]
@Mugen87 Good suggestion, thanks. I implemented it in this PR.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
I think `colorspace_fragment` and `colorspace_pars_fragment` would fit better with the rest of the shaderchunk names.

**Comment by WestLangley** [comment]
We decided to use the term "color space", not "colorspace".

Hence, the variable name became `colorSpace`, not `colorspace`.

Consequently, I chose to use `colors_space_fragment`, not `colorspace_fragment`.

Given that explanation, do you still want me to change it?

**Comment by WestLangley** [comment]
We decided to use the term "color space", not "colorspace".

Hence, the variable name became `colorSpace`, not `colorspace`.

Consequently, I chose to use `colors_space_fragment`, not `colorspace_fragment`.

Given that explanation, do you still want me to change this PR?

File: examples/jsm/loaders/3MFLoader.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ import {
 } from 'three';
 import * as fflate from '../libs/fflate.module.js';
 
-const COLOR_SPACE_3MF = SRGBColorSpace;
+const colorspace_3MF = SRGBColorSpace;
```
Comment (Mugen87) [inline examples/jsm/loaders/3MFLoader.js:24]: Can we keep these constants in upper case, please?

Comment (WestLangley) [reply examples/jsm/loaders/3MFLoader.js:24]: ugh... global replace error.

**Comment by mrdoob** [comment]
@WestLangley Thanks! 🙏

Also great to finally have a migration helper for these 🥲

**Comment by WestLangley** [comment]
This PR requires updated build files. Otherwise, the modified examples are out-of-sync with core.

**Comment by Mugen87** [comment]
e31a6d48a51a04c47281dfb330ed7f433b9228b3 🚀

**Pull request merged**